File: C++/Basic_Binary_Tree.cpp
#include <iostream>
#include <queue>
using namespace std;

struct Node {
    int data;
    Node* left;
    Node* right;
};

Node* CreateNode(int data) {
    Node* newNode = new Node();
    newNode->data = data;
    newNode->left = newNode->right = NULL;
    return newNode;
}

Node* InsertNode(Node* root, int data) {
    if (root == NULL) {
        root = CreateNode(data);
        return root;
    }

    queue<Node*> q;
    q.push(root);

    while (!q.empty()) {
        Node* temp = q.front();
        q.pop();

        if (temp->left != NULL)
            q.push(temp->left);
        else {
            temp->left = CreateNode(data);
            return root;
        }

        if (temp->right != NULL)
            q.push(temp->right);
        else {
            temp->right = CreateNode(data);
            return root;
        }
    }
    return root;
}

void inorder(Node* temp) {
    if (temp == NULL)
        return;

    inorder(temp->left);
    cout << temp->data << ' ';
    inorder(temp->right);
}

int main() {
    Node* root = NULL;
    int key;

    while (true) {
        cout << "Choose an option:\n";
        cout << "1. Insert a number into the binary tree\n";
        cout << "2. Display the inorder traversal\n";
        cout << "3. Exit\n";
        int choice;
        cin >> choice;

        switch (choice) {
            case 1:
                cout << "Enter a number to insert into the binary tree: ";
                if (!(cin >> key))
                    break;
                root = InsertNode(root, key);
                cout << "Insertion completed";
                cout << endl;
                break;
            case 2:
                cout << "Inorder traversal: ";
                inorder(root);
                cout << endl;
                break;
            case 3:
                cout << "Exiting the program\n";
                return 0;
            default:
                cout << "Invalid choice. Please try again.\n";
        }
    }

    return 0;
}

File: Python/Basic_Binary_Tree.py
from queue import SimpleQueue

class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def CreateNode(data):
    new_node = Node(data)
    return new_node

def InsertNode(root, data):
    if root is None:
        return CreateNode(data)

    q = SimpleQueue()
    q.put(root)

    while not q.empty():
        temp = q.get()

        if temp.left:
            q.put(temp.left)
        else:
            temp.left = CreateNode(data)
            return root

        if temp.right:
            q.put(temp.right)
        else:
            temp.right = CreateNode(data)
            return root

    return root

def inorder(temp):
    if temp is None:
        return

    inorder(temp.left)
    print(temp.data, end=' ')
    inorder(temp.right)

if __name__ == "__main__":
    root = None

    while True:
        print("Choose an option:")
        print("1. Insert a number into the binary tree")
        print("2. Display the inorder traversal")
        print("3. Exit")
        choice = input()

        if choice == "1":
            key = int(input("Enter a number to insert into the binary tree: "))
            root = InsertNode(root, key)
            print("Insertion completed")
            print()
        elif choice == "2":
            print("Inorder traversal: ", end='')
            inorder(root)
            print()
        elif choice == "3":
            print("Exiting the program")
            break
        else:
            print("Invalid choice. Please try again.")
